### PR TITLE
Promote DS MaxSurge to beta

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Promote DS MaxSurge to beta
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
MaxSurge is beneficial when we want 0 downtime for the DS
https://github.com/kubernetes/enhancements/pull/2665

#### Does this PR introduce a user-facing change?

```release-note
Enable MaxSurge for DS by default
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: Enable MaxSurge for DS by default
```

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
cc @smarterclayton @soltysh @janetkuo

**Comment by JamesLaverack** [comment]
xref https://github.com/kubernetes/enhancements/issues/1591

File: staging/src/k8s.io/api/apps/v1/types.go [hunk 1]
```diff
@@ -522,7 +522,7 @@ type RollingUpdateDaemonSet struct {
 	// daemonset on any given node can double if the readiness check fails, and
 	// so resource intensive daemonsets should take into account that they may
 	// cause evictions during disruption.
-	// This is an alpha field and requires enabling DaemonSetUpdateSurge feature gate.
+	// This is beta field and enabled/disabled by DaemonSetUpdateSurge feature gate.
```
Comment (ravisantoshgudimetla) [inline staging/src/k8s.io/api/apps/v1/types.go:525]: This ensures generated files have correct entries too.

**Comment by ravisantoshgudimetla** [comment]
Since DS runs serially, I expect the tests to show up at https://testgrid.k8s.io/sig-apps#gce-serial

File: test/e2e/apps/daemon_set.go [hunk 2]
```diff
@@ -491,7 +491,7 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 	})
 
 	// TODO: This test is expected to be promoted to conformance after the feature is promoted
-	ginkgo.It("should surge pods onto nodes when spec was updated and update strategy is RollingUpdate [Feature:DaemonSetUpdateSurge]", func() {
+	ginkgo.It("should surge pods onto nodes when spec was updated and update strategy is RollingUpdate", func() {
```
Comment (ravisantoshgudimetla) [inline test/e2e/apps/daemon_set.go:494]: This should help in running tests without featuregate flag

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**